client/src/Backend/GameLogic/GameManager.ts

```diff
@@ -4969,20 +4969,35 @@ class GameManager extends EventEmitter {
    */
   getTimeForMove(fromId: LocationId, toId: LocationId, abandoning = false): number {
     const from = this.getPlanetWithId(fromId);
-    if (!from) throw new Error('origin planet unknown');
-    const dist = this.getDist(fromId, toId);
+    if (!isLocatable(from)) {
+      throw new Error(`Could not find planet from (fromId: ${fromId})`);
+    }
 
-    const speed = from.speed * this.getSpeedBuff(abandoning);
+    const to = this.getPlanetWithId(toId);
+    if (!isLocatable(to)) {
+      throw new Error(`Could not find planet from (toId: ${to})`);
+    }
 
-    let deltaTime = dist / (speed / 100);
+    // NOTE: The get distance will throw if both the fromId and toId planets are not located.
+    const dist = this.getDist(fromId, toId);
 
-    // //round 1
-    // const artifact = this.getActiveArtifact(from);
-    // //MyTodo:    //all photoid travel same time
-    // if (artifact?.artifactType === ArtifactType.PhotoidCannon) deltaTime = 600;
+    // NOTE: The speed factor will always be 1 when SPACE_JUNK_ENABLED=false
```

Comment:
If we are supposed to have a speed factor adjustment when abandoning and space junk is disabled, i think we need to either:

1. Not rely on `getSpeedBuff()` method and inline the logic
2. Modify the `getSpeedBuff()` method.